prepare-image.sh

```diff
@@ -2,28 +2,54 @@
 
 set -euxo pipefail
 
+IRONIC_PKG_LIST=/tmp/ironic-${INSTALL_TYPE}-list.txt
+
 echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
 # Tell RPM to skip installing documentation
 echo "tsflags=nodocs" >> /etc/dnf/dnf.conf
 
 dnf install -y python3 python3-requests epel-release 'dnf-command(config-manager)'
 dnf config-manager --set-disabled epel
-curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/plugins/module_utils/tripleo_repos/main.py | python3 - -b master current-tripleo
-dnf upgrade -y
+
+# RPM install #
+if [[ $INSTALL_TYPE == "rpm" ]]; then
+    curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/plugins/module_utils/tripleo_repos/main.py | python3 - -b master current-tripleo
+    dnf upgrade -y
+    xargs -rtd'\n' dnf install -y < $IRONIC_PKG_LIST
+fi
+
+# SOURCE install #
+if [[ $INSTALL_TYPE == "source" ]]; then
+    dnf upgrade -y
+    dnf install -y python3-pip python3-devel gcc git-core
+    pip3 install --upgrade pip
```

Comment:
I'd rather avoid an unconditional upgrade of pip, we've had enough issues with it upstream